### PR TITLE
feat: add gitlab openssf scorecards

### PR DESCRIPTION
**Pull request by adam-moss**
## Changes

Adds OpenSSF scorecard reporting for GitLab.

## Context

Adding following testing as part of https://github.com/renovatebot/renovate/discussions/25125#discussioncomment-7769281

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [X] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/security.ts [hunk 1]
```diff
@@ -5,10 +5,10 @@ export const presets: Record<string, Preset> = {
     description: 'Show OpenSSF badge on pull requests.',
     packageRules: [
       {
-        matchSourceUrlPrefixes: ['https://github.com/'],
+        matchSourceUrlPrefixes: ['https://github.com/', 'https://gitlab.com'],
         prBodyDefinitions: {
           OpenSSF:
-            '[![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/{{sourceRepo}}/badge)](https://securityscorecards.dev/viewer/?uri=github.com/{{sourceRepo}})',
+            '[![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/{{platform}}.com/{{sourceRepo}}/badge)](https://securityscorecards.dev/viewer/?uri={{platform}}.com/{{sourceRepo}})',
```
Comment (secustor) [inline lib/config/presets/internal/security.ts:11]: `platform` returns the platform against which Renovate is run and not where the source of the dependency is located

**Review by secustor — CHANGES_REQUESTED** [review]

**Comment by adam-moss** [comment]
Screenshot 2023-12-06 at 17 47 59


Tested against a private repository.  Seems the OpenSSF api has a challenge with /group/subgroup/project urls.  I wonder if there is something nicer we could do rather than having a broken image link?

Example:
https://gitlab.com/gitlab-org/ruby/gems/gitlab-triage
https://api.securityscorecards.dev/projects/gitlab.com/gitlab-org/ruby/gems/gitlab-triage/badge

**Comment by adam-moss** [comment]
Indeed, I have raised https://github.com/ossf/scorecard-webapp/issues/511

**Comment by rarkins** [comment]
What's the next step here?

**Comment by secustor** [comment]
It's not actionable by us right now and blocked by OSSF scorecards API server implementation

**Comment by rarkins** [comment]
OK, please feel free to reopen this as soon as it's actionable. We have 60+ open PRs right now so would like to close anything non-actionable and stalled.

**Pull request closed without merging**